src/coreclr/jit/optimizer.cpp

```diff
@@ -604,6 +604,8 @@ void Compiler::optClearLoopIterInfo()
         loop.lpConstInit = -1;
         loop.lpTestTree  = nullptr;
     }
+
+    return PhaseStatus::MODIFIED_NOTHING;
```

Comment:
Well, `NOTHING` is a misnomer. So maybe you could at least add a comment at the definition of `MODIFIED_NOTHING` that says what counts as something and what counts as nothing?